File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/impl/StatusClientImpl.java
package com.github.seratch.jslack.api.status.v2.impl;

import com.github.seratch.jslack.api.status.v2.StatusApiException;
import com.github.seratch.jslack.api.status.v2.StatusClient;
import com.github.seratch.jslack.api.status.v2.model.CurrentStatus;
import com.github.seratch.jslack.api.status.v2.model.SlackIssue;
import com.github.seratch.jslack.common.http.SlackHttpClient;

import java.io.IOException;
import java.util.List;

public class StatusClientImpl implements StatusClient {

    private String endpointUrlPrefix = ENDPOINT_URL_PREFIX;
    private final SlackHttpClient slackHttpClient;

    public StatusClientImpl(SlackHttpClient slackHttpClient) {
        this.slackHttpClient = slackHttpClient;
    }

    @Override
    public String getEndpointUrlPrefix() {
        return this.endpointUrlPrefix;
    }

    @Override
    public void setEndpointUrlPrefix(String endpointUrlPrefix) {
        this.endpointUrlPrefix = endpointUrlPrefix;
    }

    @Override
    public CurrentStatus current() throws IOException, StatusApiException {
//        Response response = slackHttpClient.get(endpointUrlPrefix + "current", null, null);
//        String body = response.body().string();
//        slackHttpClient.runHttpResponseListeners(response, body);
//        if (response.isSuccessful()) {
//            return GsonFactory.createSnakeCase(slackHttpClient.getConfig()).fromJson(body, CurrentStatus.class);
//        } else {
//            throw new StatusApiException(response, body);
//        }
        throw new IllegalStateException("Not supported.");
    }

    @Override
    public List<SlackIssue> history() throws IOException, StatusApiException {
//        Response response = slackHttpClient.get(endpointUrlPrefix + "history", null, null);
//        Type listType = new TypeToken<ArrayList<SlackIssue>>() {
//        }.getType();
//        String body = response.body().string();
//        slackHttpClient.runHttpResponseListeners(response, body);
//        if (response.isSuccessful()) {
//            return GsonFactory.createSnakeCase(slackHttpClient.getConfig()).fromJson(body, listType);
//        } else {
//            throw new StatusApiException(response, body);
//        }
        throw new IllegalStateException("Not supported.");
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/impl/StatusClientImpl.kt
package work.socialhub.kslack.api.status.impl

import work.socialhub.khttpclient.HttpRequest
import work.socialhub.kslack.Slack
import work.socialhub.kslack.api.methods.SlackApiException
import work.socialhub.kslack.api.methods.helper.JsonHelper.fromJson
import work.socialhub.kslack.api.status.StatusClient
import work.socialhub.kslack.api.status.model.CurrentStatus
import work.socialhub.kslack.api.status.model.SlackIssue

class StatusClientImpl : StatusClient {

    override var endpointUrlPrefix: String? = Slack.STATUS_URL_PREFIX

    override suspend fun current(): CurrentStatus {
        val response = HttpRequest()
            .url("${endpointUrlPrefix}current")
            .get()

        if (response.status in 200..<299) {
            return fromJson(response.stringBody)
        }
        throw SlackApiException(response)
    }

    override suspend fun history(): Array<SlackIssue> {
        val response = HttpRequest()
            .url("${endpointUrlPrefix}history")
            .get()

        if (response.status in 200..<299) {
            return fromJson(response.stringBody)
        }
        throw SlackApiException(response)
    }
}